pkg/porter/publish.go

```diff
@@ -88,17 +87,33 @@ func (p *Porter) publishFromFile(opts PublishOptions) error {
 			return errors.Wrapf(err, "unable to set invocation image name from tag %q", tag)
 		}
 	} else {
+		// If the manifest file is the default/user-supplied manifest,
+		// hot-swap in Porter's canonical translation (if exists) from
+		// the .cnab/app directory, as there may be dynamic overrides for
+		// the name and version fields to inform invocation image naming.
+		canonicalExists, err := p.FileSystem.Exists(build.LOCAL_MANIFEST)
+		if err != nil {
+			return err
+		}
+
+		if opts.File == config.Name && canonicalExists {
```

Comment:
Why the check for if opts.File == "porter.yaml"?

Reply:
My thinking was that if `opts.File` was different than the default/`config.Name` manifest, we'd want to skip this section and quickly drop down to the check to see if the bundle is up-to-date and trigger a re-build, which will happen assuming the manifest file provided by `opts.File` is different from the default.  However, further updates would be needed to fully support publishing from a manifest file other than the default, including updating the `generateInternalManifest` function added in this PR to accept a filepath variable indicating the original manifest (right now, we are hard-coding it to `config.Name`).  Shall I start focusing on this flow and make updates as needed?

Reply:
Since this PR isn't about supporting loading the manifest from anything other than ./porter.yaml (and we both agree that loading from another file _does not work right now) I say don't add anything for that use case. We can follow-up on it later when/if it's necessary.

Let's not add anything else to this PR that isn't related to the main issue cuz this is a chonky PR. 😀

Reply:
Ok, check removed.